runtime/src/vm/opcode.rs

```diff
@@ -3063,9 +3063,123 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
   });
 
   // "Special" opcodes
-  io!("waitop", |args, hand_mem, _, _| {
-    let ms = hand_mem.read_fixed(args[0]) as u64;
-    return Box::pin(delay_for(Duration::from_millis(ms)));
+  io!("waitop", |args, mem| {
+    let fut = async move {
+      let hand_mem = mem.read().await;
+      let ms = hand_mem.read_fixed(args[0]) as u64;
+      drop(hand_mem); // drop read lock
+      delay_for(Duration::from_millis(ms)).await;
+    };
+    return Box::pin(fut);
+  });
+  io!("httpget", |args, mem| {
+    let fut = async move {
```

Comment:
bol

Reply:
![futbol](https://media2.fdncms.com/eastbayexpress/imager/u/original/17703015/soccer-ball-ss-img.jpg)

Reply:
yes! ⚽️ definitely comes to mind before 🏈 for me

Reply:
Well, in this case, because the `fut`ure is not a `foot`ure, otherwise definitely the latter ;)